src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -1629,6 +1635,30 @@ private CompressedPagesResult PrepareToWriteToJournal(LowLevelTransaction tx, IP
             return prepreToWriteToJournal;
         }
 
+        private static int _pagesIn1Mb = Constants.Size.Megabyte / Constants.Storage.PageSize;
+
+        /// <summary>
+        /// The idea of this function is to calculate page sizes that will cause less fragmentation in 32 bit mode
+        /// for allocation smaller than 1MB we will allocate the next power of 2
+        /// for allocation larger than 1MB we will alligned them to be MB alligned 
+        /// </summary>
+        /// <param name="pagesRequired"></param>
+        /// <returns></returns>
+        private static int AdjustPagesRequiredFor32Bits(int pagesRequired)
+        {
+            var bytes = pagesRequired * Constants.Storage.PageSize;
+            if (bytes < Constants.Size.Megabyte / 2)
+            {
+                pagesRequired = Bits.NextPowerOf2(bytes / Constants.Storage.PageSize + bytes % Constants.Storage.PageSize == 0 ? 0 : 1);
```

Comment:
```suggestion
                pagesRequired = Bits.NextPowerOf2(bytes) / Constants.Storage.PageSize ;
```

Shouldn't this be just this? There can't be any remainder here.